django/librarian/templates/librarian/forms/data_source.html

```diff
@@ -7,7 +7,7 @@
   {% url 'librarian:modal_delete_data_source' data_source_id=form.instance.id as delete_url %}
 {% else %}
   {% url 'librarian:modal_create_data_source' library_id=form.library.value as form_action_url %}
-  {% url 'librarian:modal_library_list' as cancel_url %}
+  {% url 'librarian:modal_datasource_list' library_id=form.library.value as cancel_url %}
```

Comment:
Ah, thanks for finding this. I think you could just use `librarian:modal_edit_library` instead though.